### ISSUE TITLE
Create a "https-verified" installation

### ISSUE THREAD

**Issue by thekid** [xp-runners/reference#29]
Add a oneliner shell script, something like Docker ("wget -qO- https://get.docker.com/ | sh") or Composer ("curl -sS https://getcomposer.org/installer | php").

**Comment by thekid** [comment]
Target: https://bintray.com/xp-runners/generic/xp-runners/view

**Comment by thekid** [comment]
Build: https://travis-ci.org/xp-runners/reference/builds/105511431

**Comment by thekid** [comment]
Needed to give up on the self-extracting script idea, this does not work well with STDIN.

**Comment by thekid** [comment]
``` sh
$ curl -sSL https://bintray.com/artifact/download/xp-runners/generic/setup-7.0.0.sh | bash
```

A successful install (Ubuntu):

![https-verified-install](https://cloud.githubusercontent.com/assets/696742/12659203/cd7bd7f6-c60d-11e5-9972-0b0b020b63b8.png)

A successful install (Cygwin):

![https-verified-install-cygwin](https://cloud.githubusercontent.com/assets/696742/12659524/86d63d80-c60f-11e5-9a87-ec35d9fed466.png)

Here's what happens when we run on Linux and don't have Mono installed:

![https-verified-install-errors](https://cloud.githubusercontent.com/assets/696742/12659401/dee68e18-c60e-11e5-8678-e9cb4b972a14.png)

When we're on Ubuntu and Debian, we know how to fix missing dependencies:

![https-verified-install-autofix](https://cloud.githubusercontent.com/assets/696742/12660868/7f6dddc6-c616-11e5-9fa9-a2f1773df04c.png)

**Comment by thekid** [comment]
![https-verified-install-errors](https://cloud.githubusercontent.com/assets/696742/12659401/dee68e18-c60e-11e5-8678-e9cb4b972a14.png)

**Comment by thekid** [comment]
For detecting OS and distribution, I've found http://unix.stackexchange.com/questions/6345/how-can-i-get-distribution-name-and-version-number-in-a-simple-shell-script useful.

**Comment by thekid** [comment]
On Windows, we'll currently suggest [Chocolatey](https://chocolatey.org/) as a possible fix for missing PHP runtime:

![https-verified-install-choco](https://cloud.githubusercontent.com/assets/696742/12662404/860cf0c2-c620-11e5-937b-6f56973461b4.png)

We could download and extract PHP ourselves on Windows instead of relying on Chocolatey being available:
- [x] Scrape http://windows.php.net/downloads/releases/
- [x] Find newest nts-version
- [x] Download ZIP, extract to same directory

``` sh
$ curl http://windows.php.net/downloads/releases/ \
  | grep -Eo 'php-[0-9.]+-nts-Win32-[VCx0-9-]+.zip' \
  | sort -rn \
  | uniq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  7614  100  7614    0     0  34572      0 --:--:-- --:--:-- --:--:-- 35915
php-7.0.2-nts-Win32-VC14-x86.zip
php-7.0.2-nts-Win32-VC14-x64.zip
php-5.6.17-nts-Win32-VC11-x86.zip
php-5.6.17-nts-Win32-VC11-x64.zip
php-5.5.31-nts-Win32-VC11-x86.zip
php-5.5.31-nts-Win32-VC11-x64.zip
php-5.4.45-nts-Win32-VC9-x86.zip
```

To determine whether to use 64 or 32 bit versions, use `uname -a`, yield x86_64 here.

**Comment by thekid** [comment]
Added two command line flags:
- `-f`: Ignore if checks for mono / .NET and PHP installations fail
- `-u`: Unattended mode - do not ask if to apply auto-fixes, exit instead.

**Comment by thekid** [comment]
Hrm, [select is not in Ubuntu's default shell, dash](https://wiki.ubuntu.com/DashAsBinSh#select), causing:

``` sh
$ sh target/setup-7.0.0.sh
target/setup-7.0.0.sh: 71: target/setup-7.0.0.sh: Syntax error: "do" unexpected (expecting "}")
```

Here are the "bashisms" we're using:

``` sh
vagrant@vagrant-ubuntu-wily-64:/devel/runners/reference$ checkbashisms setup.sh.in
possible bashism in setup.sh.in line 21 (echo -e):
  echo -ne "> $message...\r"
possible bashism in setup.sh.in line 27 (should be 'b = a'):
    if [ Warning == $type ] ; then
possible bashism in setup.sh.in line 32 (should be 'b = a'):
    if [ No == "$fix" ] || [ Unattended == $MODE ] ; then
possible bashism in setup.sh.in line 39 (read with option other than -r):
    read -p 'Should we do this now (y/N)? ' answer
possible bashism in setup.sh.in line 40 (should be 'b = a'):
    if [ y == "$answer" ] ; then
possible bashism in setup.sh.in line 56 (type):
  if type curl >/dev/null 2>&1; then
possible bashism in setup.sh.in line 58 (type):
  elif type wget >/dev/null 2>&1; then
possible bashism in setup.sh.in line 71 ('select' is not POSIX):
  select release in $(cat $TEMPFILE); do
possible bashism in setup.sh.in line 95 (echo -e):
  echo -ne "> Configuration\r"
possible bashism in setup.sh.in line 108 (echo -e):
echo -e "\e[32;1mXP Runners setup (@VERSION@)\e[0m"
possible bashism in setup.sh.in line 125 (should be 'b = a'):
if [ 'Linux' == "$OS" ] ; then OS=$(lsb_release -si) ; fi
```

**Comment by thekid** [comment]
> Syntax error: "do" unexpected (expecting "}")

Fixed in xp-runners/reference@79aea2e38758bfdd1d13d5a0bc3c43d5fbdabd0a

> [...and more..] "bashisms" we're using

Also fixed: echo -e and echo -n were replaced by `printf`, type with `which` and == with a single `=`.  Also had to rewrite \e to `\033`, although the above script didn't complain about it, it was still broken.

**Comment by thekid** [comment]
PHP is now bundled as auto-fix as suggested above.

![https-verified-install-bunde-php](https://cloud.githubusercontent.com/assets/696742/12698874/10857c90-c7a9-11e5-85ca-9575b3a58b38.png)
